### ISSUE TITLE
mat-chip-list should be grid instead of listbox

### ISSUE THREAD

**Issue by stevenyxu** [angular/components#13499]
#### Bug, feature request, or proposal:
Update mat-chip-list from ARIA listbox to grid. Slight tweak of attributes and keyboard shortcuts for grid semantics.

#### What is the expected behavior?
mat-chip-list should have ARIA `role=grid`.

#### What is the current behavior?
mat-chip-list is presented with `role=listbox`.

#### What are the steps to reproduce?
The chips with input example on the demo page shows the full range of behavior. If you open this and navigate to the list of added chips using NVDA/JAWS virtual cursor, the SR fails to announce the current selection logically.

https://stackblitz.com/angular/xvvkqgrrebo?file=app%2Fchips-input-example.ts

#### What is the use-case or motivation for changing an existing behavior?
mat-chip-list semantics (browse, select, delete) do not match role=listbox semantics (select one option on focus). mat-chip-list has been presented with `role=listbox` since its inception (edf5cdf). It was later tweaked slightly (dc76c09) to only have the role when it had options. This is an imprecise usage of the listbox pattern https://www.w3.org/TR/wai-aria-practices/#Listbox which is something that contains only selectable or multiselectable options and groups of options with no finer grained semantics within each option (selecting the chip vs. pressing the delete icon, key-value distinction for filter chips). Most importantly, SRs treat listboxes as a widget to facilitate selection, and our tester reports that NVDA/JAWS boots the user to "forms mode" when a listbox is engaged using the virtual cursor, leading to a confusing experience.

Based on the https://www.w3.org/TR/wai-aria-practices/#no_aria_better_bad_aria principle, it seems like it might be safer to go with a plainer grid role. In fact, "pill list" is one of the exemplar use cases for grid in the WAI best practices guide https://www.w3.org/TR/wai-aria-practices/examples/grid/LayoutGrids.html#ex2_label.

I'm not attached to a specific role, so you'd like more info from our testers of exactly what's wrong, please let me know and I'll have them join the thread.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
All platforms affected. The bug presents in NVDA/JAWS, which have specific form-like behavior when entering listboxes.

#### Is there anything else we should know?
N/A

**Comment by jelbourn** [comment]
This is on our roadmap; we did some experimenting with switching over to `grid` already (#11465), but there were some fundamental issues causing the change to be too breaking for existing applications inside Google. The change will come when we do a larger overhaul of the chips component.

**Comment by vanessanschmitt** [comment]
Could we get a status update on when we could expect a chips component overhaul? No worries if it's not currently prioritized. But we are evaluating whether we need to find a workaround for some of the chip list accessibility issues, and want to make sure we don't put in unnecessary work if a chips rewrite is coming soon.

**Comment by jelbourn** [comment]
It will probably be either Q4 or Q1 2020 when this would end up being available. The overhaul I have in mind would be making the chips more flexible to support different interaction patterns, something kind of like
```html
<mat-chip-listbox>
  <mat-chip-option> ... </>
</>

<mat-chip-grid>
  <mat-chip-cell> ... </>
</>


<mat-chip-radiogroup>
  <mat-chip-radiobutton> ... </>
</>

<mat-chip> ... </> <!-- no role, no aria attrs, DIY -->
```
